### PR TITLE
Fix still non getentropy platform builds

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by devnexen** [comment]
sorry here we go again but it fails on netbsd (at least).

**Comment by mjp41** [comment]
@devnexen, sorry I am confused by your NetBSD comment.  Is this PR fixing NetBSD?  Or is NetBSD broken even with this change?

I prefer the guarded `| SNMALLOC_PLATFORM_HAS_GETENTROPY` you are proposing, but I don't see what this fixes? Could you give some extra guidance on your proposed change.

**Comment by devnexen** [comment]
No pb :-) Yes it is for fixing NetBSD build, see SNMALLOC_PLATFORM_HAS_GETENTROPY was always defined regardless (`0` or `Entropy`) thus went back to square one. Hopefully it will solve for all :-)

File: CMakeLists.txt [hunk 1]
```diff
@@ -178,8 +178,6 @@ endif()
 
 if(SNMALLOC_PLATFORM_HAS_GETENTROPY)
   target_compile_definitions(snmalloc_lib INTERFACE -DSNMALLOC_PLATFORM_HAS_GETENTROPY=Entropy)
```
Comment (mjp41) [inline CMakeLists.txt:180]:
```suggestion
  target_compile_definitions(snmalloc_lib INTERFACE -DSNMALLOC_PLATFORM_HAS_GETENTROPY)
```

File: src/pal/pal_posix.h [hunk 2]
```diff
@@ -126,8 +120,11 @@ namespace snmalloc
      * POSIX systems are assumed to support lazy commit. The build system checks
      * getentropy is available, only then this PAL supports Entropy.
      */
-    static constexpr uint64_t pal_features =
-      LazyCommit | SNMALLOC_PLATFORM_HAS_GETENTROPY;
+    static constexpr uint64_t pal_features = LazyCommit
+#if defined(SNMALLOC_PLATFORM_HAS_GETENTROPY)
+      | SNMALLOC_PLATFORM_HAS_GETENTROPY
```
Comment (mjp41) [inline src/pal/pal_posix.h:125]:
```suggestion
      | Entropy
```

**Review by mjp41 — CHANGES_REQUESTED** [review]
Okay, I see
https://github.com/microsoft/snmalloc/blob/aefd149f0ace9307b2fef2e8d803472d2b5a3622/src/pal/pal_posix.h#L301-L306
is still being compiled causing the error.  I think given this change, I would sooner we not expose `Entropy` element of the `enum` to `CMake`.

File: src/pal/pal_posix.h (hunk 2 above)
Comment (devnexen) [reply src/pal/pal_posix.h:125]: ok makes sense was not sure if you wanted to keep `Entropy`

**Review by mjp41 — APPROVED** [review]
LGTM

**Pull request merged**